app/models/article.rb

```diff
@@ -552,6 +555,30 @@ def validate_collection_permission
     errors.add(:collection_id, "must be one you have permission to post to")
   end
 
+  def validate_co_authors
+    return unless co_author_ids
+    return unless co_author_ids.include?(user_id)
+
+    errors.add(:co_author_ids, "must not be the same user as the author")
+  end
+
+  def validate_co_authors_must_not_be_the_same
+    return if co_author_ids.blank?
+
+    return if co_author_ids.uniq.count == co_author_ids.count
+
+    errors.add(:base, "co-author IDs must be unique")
+  end
+
+  def validate_co_authors_exist
+    return unless co_author_ids
+
+    valid_co_authors = User.where(id: co_author_ids).count == co_author_ids.count
+    return if valid_co_authors
+
+    errors.add(:co_author_ids, "must be valid user IDs containing only numbers")
```

Comment:
I'm a bit confused about this validation + error message.

We check this: `User.where(id: co_author_ids).count == co_author_ids.count`, which checks to see that the `count` of Users that exist in the database with the `co_author_ids` of the article is the same as the `count` of the `co_author_ids` on the Article. But I'm not sure how that is ensuring that `co_author_ids` are _only_ going to be numbers? We could instead validate that the `co_author_ids` are actually only numbers here if we wanted to use this error message.

Reply:
Soooo, I kind of just got ahead of myself and have since changed this by removing the "containing only numbers" portion of the error message. 😅 😬